converter/src/test/java/gov/cms/qpp/conversion/ConverterTest.java

```diff
@@ -50,17 +50,17 @@
 	@Test(expected = org.junit.Test.None.class)
 	public void testValidQppFile() {
 		Path path = Paths.get("../qrda-files/valid-QRDA-III-latest.xml");
-		Converter converter = new Converter(new PathQrdaSource(path));
+		Converter converter = new Converter(new PathSource(path));
 
 		converter.transform();
 		//no exception should be thrown, hence explicitly stating the expected exception is None
 	}
 
 	@Test(expected = org.junit.Test.None.class)
-	public void testValidQppStream() {
+	public void testValidQppStream() throws IOException {
 		Path path = Paths.get("../qrda-files/valid-QRDA-III-latest.xml");
 		Converter converter = new Converter(
-				new InputStreamSupplierQrdaSource(path.toString(), () -> XmlUtils.fileToStream(path)));
+				new InputStreamSupplierSource(path.toString(), () -> XmlUtils.fileToStream(path), Files.size(path)));
```

Comment:
Is there a reason you favored this constructor over the 2 arg constructor?

Reply:
Because the two argument constructor pulls the `InputStream` into memory to compute the size.  This three argument constructor takes the size and just uses it without having to pull the entire source into memory.